src/Corax/IndexFieldsMapping.cs

```diff
@@ -99,31 +102,30 @@ public bool TryGetByFieldId(int fieldId, out IndexFieldBinding binding)
         return FieldsById.TryGetValue(fieldId, out binding);
     }
 
-    public IndexFieldBinding GetByFieldName(string fieldName)
+    public IndexFieldBinding GetByFieldName(ByteStringContext context, string fieldName)
     {
         // This method is a convenience method that should not be used in high performance sections of the code.
-        using var _ = Slice.From(Context, fieldName, out var str);
+        using var _ = Slice.From(context, fieldName, out var str);
         return Fields[str];
     }
 
-    public bool TryGetByFieldName(string fieldName, out IndexFieldBinding binding)
+    public bool TryGetByFieldName(ByteStringContext context, string fieldName, out IndexFieldBinding binding)
```

Comment:
@ayende  This was the real issue. We shared a context between threads. Now I've changed the API to make sure we cannot use shared context (except adding something to binding)

Reply:
Can we create an *explicit* difference between them?
So we'll have a Builder instance that is mutatable, and a read only one that cannot change